### DISCUSSION TITLE
url_to with params not rendering params

### DISCUSSION THREAD

**Discussion by janlimpens** [mojolicious/mojo#2003] [Q&A] [answered]
* Mojolicious version: 9.26
* Perl version: 5.36
* Operating system: Debian bullseye

I have a mojolicious lite app and in it, I have this route:
```get 'some_route/<page:num>' => sub($c) { ... };```

In the template, I try to call it like this:
```
%= url_for('some_route', {page => 1});
# or
%= url_for('some_route', page => 1);
```

Which renders
```some_route```

and I think, it should render 
```some_route/1```

Am I misunderstanding anything here? I am following 
https://docs.mojolicious.org/Mojolicious/Controller#url_for
and
https://docs.mojolicious.org/Mojolicious/Guides/Routing#Optional-placeholders

**Comment by kraih** [comment, marked as answer]
https://docs.mojolicious.org/Mojolicious/Guides/Routing#Named-routes

**Reply by janlimpens** [reply]
Thanks for the response! 

Ok, even if 

> You can assign a name with ["name" in Mojolicious::Routes::Route](http://docs.mojolicious.org/Mojolicious/Routes/Route#name), or let the router generate one automatically, which would be equal to the route itself without non-word characters, custom names have a higher precedence though.

seems to imply that naming is not necessary.

How would I do this in a scenario, where I just have one Lite .pl file with a bunch of routes like 

`get 'something/:param' => sub ($c) { ... } => 'some_template';`

I currently have no setup method as outlined in
https://docs.mojolicious.org/Mojolicious/Guides/Routing#Minimal-route
and it seems to be a piece of a Mojolicious 'not' Lite app... At present, if possible, I would want to avoid this growth.

**Reply by janlimpens** [reply]
I (think, I) found it out!

`get 'something/:param' => sub ($c) {  $c->render(template => 'some_template');  } => 'route_name';`

Could be a bit clearer in the documentation, though :)

**Reply by kraih** [reply]
You did not check to see what the auto-generated name actually was though. 😉  https://docs.mojolicious.org/Mojolicious/Guides/Routing#Introspection

**Reply by janlimpens** [reply]
Yeah, I used a nonexistent route in the browser and found out, by the error page, that I had several routes sharing the same name. 

That was because I had used that tailing `=>'something'` to assign the same template to several actions. Once that was cleared up, I knew, where to look.

**Reply by guest20** [reply]
It might be helpful for devs if `url_for` died when given it a bogus route name (when `app->mode eq 'development'`)?

Right now you just get silent bug (an a with an empty href, etc) which "looks ok" if you don't actually click it each time you make a change.
